<launch>
  <arg name="pwm_path" value="/Arm_Board_HW/PWM_Channels/PulseTimeControl/" />
  <arg name="simulated" default="false" />
  <group ns="Arm_Board_HW" unless="$(arg simulated)">
    <node name="PCA9685_PWM_HAT" pkg="mavric" type="PCA9685_PWM_HAT.py" respawn="true">
      <param name="frequency" value="50" />
      <!-- Determine as SetPeriod/RealPeriod -->
      <param name="clk_error" value="1.08" />
    </node>
  </group>

  <group ns="Arm">
	<node name="Arm_Control" pkg="mavric" type="Arm_S.py" respawn="true" output="screen">
		<remap from="Arm_ShoulderRot" to="Command_ShoulderRot" />
		<remap from="Arm_ShoulderPitch" to="Command_ShoulderPitch" />
		
		<remap from="Arm_ElbowPitch" to="Command_ElbowPitch" />
		
		<remap from="Arm_WristRot" to="Command_WristRot" />
		<remap from="Arm_WristPitch" to="Command_WristPitch" />
		
		<remap from="Arm_ClawActuation" to="Command_ClawActuation" />
		
		<remap from="ShoulderRot" to="$(eval arg('pwm_path') + 'CH15')" />
		<remap from="ShoulderPitch" to="$(eval arg('pwm_path') + 'CH14')" />
		
		<remap from="ElbowPitch" to="$(eval arg('pwm_path') + 'CH13')" />
		
		<remap from="WristRot" to="$(eval arg('pwm_path') + 'CH12')" />
		<remap from="WristPitch" to="$(eval arg('pwm_path') + 'CH11')" />
		
		<remap from="ClawActuation" to="$(eval arg('pwm_path') + 'CH10')" />
		<param name="Range" value="0.3" />
		
		<param name="ShoulderRot/Scale" value="1" />
		<param name="ShoulderPitch/Scale" value="1" />
		<param name="ElbowPitch/Scale" value="3.333" />
		<param name="WristRot/Scale" value="1" />
		<param name="WristPitch/Scale" value="1" />
		<param name="ClawActuation/Scale" value="3.333" />
	</node>
  </group>
</launch>
